sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/AnalysisHelper.scala

```diff
@@ -94,6 +94,8 @@ trait AnalysisHelper extends QueryPlan[LogicalPlan] { self: LogicalPlan =>
             rule.applyOrElse(afterRuleOnChildren, identity[LogicalPlan])
           }
         }
+        newNode.copyTagsFrom(this)
```

Comment:
This exists in `transformUp`, but not in `resolveOperatorsUp` -  was the difference intentional or unintentional? Without the tags, the metadata cannot be resolved properly (`isMetadataCol` is always false).

Reply:
I think it's a mistake.